### PR TITLE
add tests with ASAN to node CI

### PR DESCRIPTION
**Pull request by satoren**

### CODE REVIEW

File: .github/workflows/mediasoup-node.yaml [hunk 1]
```diff
@@ -11,12 +11,22 @@ concurrency:
...
           - os: ubuntu-20.04
             node: 18
           - os: ubuntu-22.04
             node: 20
+          - os: ubuntu-24.04
+            node: 20
+            meson_args: '-Db_sanitize=address'
+          - os: ubuntu-24.04
+            node: 20
+            meson_args: '-Db_sanitize=undefined'
```
Comment (ibc) [inline .github/workflows/mediasoup-node.yaml:26]: Is `-Db_sanitize=undefined` the same as not passing `-Db_sanitize` argument?

File: .github/workflows/mediasoup-node.yaml [hunk 2]
```diff
@@ -11,12 +11,22 @@ concurrency:
...
+          - os: ubuntu-24.04
+            node: 20
+            meson_args: '-Db_sanitize=thread'
```
Comment (ibc) [inline .github/workflows/mediasoup-node.yaml:29]: What is the rationale for having 3 separate matrix entries with different `-Db_sanitize` options? Why not all together into a single one?

**Review by ibc — COMMENTED** [review]
Looks good but I have questions.

File: .github/workflows/mediasoup-node.yaml (hunk 1 above)
Comment (satoren) [reply .github/workflows/mediasoup-node.yaml:26]: No, this is an undefined [behavior sanitizer](https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)

File: .github/workflows/mediasoup-node.yaml (hunk 2 above)
Comment (satoren) [reply .github/workflows/mediasoup-node.yaml:29]: Only one sanitizer can be enabled at a time.
For example, thread sanitizer and address sanitizer cannot be enabled at the same time.

File: .github/workflows/mediasoup-node.yaml [hunk 3]
```diff
@@ -11,12 +11,22 @@ concurrency:
 jobs:
   ci:
     strategy:
+      fail-fast: false
```
Comment (ibc) [inline .github/workflows/mediasoup-node.yaml:14]: While this is good while writing a PR we prefer not to have this once merged so I will remove this once we merge this PR.

**Review by ibc — APPROVED** [review]

File: .github/workflows/mediasoup-node.yaml (hunk 2 above)
Comment (ibc) [reply .github/workflows/mediasoup-node.yaml:29]: Then why this?

```
clang++ -fsanitize=signed-integer-overflow,null,alignment -fno-sanitize-recover=null -fsanitize-trap=alignment a.cc
```

https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html

**Comment by ibc** [comment]
@satoren wanna help in this similar PR for worker tests? https://github.com/versatica/mediasoup/pull/1416

File: .github/workflows/mediasoup-node.yaml (hunk 2 above)
Comment (ibc) [reply .github/workflows/mediasoup-node.yaml:29]: I'm testing `-fsanitize='undefined,address,thread` and it definitely works. So honestly I don't think that we need to run just one at the same time.

Comment (ibc) [reply .github/workflows/mediasoup-node.yaml:29]: ok, but the I read this:

https://stackoverflow.com/a/68907718

**Pull request merged**